src/python/WMCore/Storage/Execute.py

```diff
@@ -96,48 +23,14 @@ def runCommandWithOutput(command):
     """
     # capture stdout and stderr from command
     child = Popen(command, shell=True, bufsize=1, stdin=PIPE, stdout=PIPE, stderr=PIPE, close_fds=True)
-    child.stdin.close()  # don't need to talk to child
-    outfile = child.stdout
-    outfd = outfile.fileno()
-    errfile = child.stderr
-    errfd = errfile.fileno()
-    makeNonBlocking(outfd)  # don't deadlock!
-    makeNonBlocking(errfd)
-    outeof = erreof = 0
-    output = ''
-    while True:
-        ready = select.select([outfd, errfd], [], [])  # wait for input
-        if outfd in ready[0]:
-            try:
-                outchunk = outfile.read()
-            except Exception as ex:
-                msg = "Unable to read stdout chunk... skipping"
-                print(msg)
-                outchunk = ''
-            if outchunk == '': outeof = 1
-            sys.stdout.write(outchunk)
-            output += outchunk
-        if errfd in ready[0]:
-            try:
-                errchunk = errfile.read()
-            except Exception as ex:
-                msg = "Unable to read stderr chunk... skipping"
-                print(msg, str(ex))
-                errchunk = ""
-            if errchunk == '': erreof = 1
-            output += errchunk
-            sys.stderr.write(errchunk)
-        if outeof and erreof: break
-        select.select([], [], [], .1)  # give a little time for buffers to fill
 
-    err = child.wait()
-    if os.WIFEXITED(err):
-        err = os.WEXITSTATUS(err)
-    elif os.WIFSIGNALED(err):
-        err = os.WTERMSIG(err)
-    elif os.WIFSTOPPED(err):
-        err = os.WSTOPSIG(err)
-    return err, output
+    retCode = child.wait()
+    sout, serr = child.communicate()
+
+    # If child is terminated by signal, err will be negative value. (Unix only)
+    sigStr = "Terminated by signal %s\n" % -retCode if retCode < 0 else ""
```

Comment:
Are you making it positive because we check (somewhere else) whether return code is > 0?
Unless I misunderstood you, I'm in favor of not changing the exit code.